### DISCUSSION TITLE
Notifiarr notifications support

### DISCUSSION THREAD

**Discussion by phutur1st** [thinkst/opencanary#369] [Feature Requests]
Hello!

I've added support locally for [notifications via Notifiarr's passthrough integration](https://github.com/phutur1st/opencanary/tree/feature-notifiarr-notification-handler). I'm considering a pull request once tweaked for public consumption, however I want to make sure the contribution delivers the best value to this project.

Included in the Notifiarr handler is options to drop fields from the json payload if they are _null_, _-1,_ or "_Missing Value_."
Additionally, it includes the option to choose which combination of time values to include in the payload (ie _utc_time, local_time, local_time_adjusted_.) I felt this approach, which dynamically removes fields not used for a given event was more practical than providing a user entered array/map of fields to include per logtype. It also iterates the logtype map and provides human readable names for the logtype field, replacing **2000** with **2000: FTP Login Attempt** for a more descriptive notification.

Before proceeding too much further, I wanted to gauge interest in moving the method for configuring/dropping null values and setting desired time fields out of the Notifiarr handler class, and into logger.py et al. This would theoretically make it easier for other handlers such as Slack or a new MS Teams method to utilize the field dropper method.

Of course if this goes stale, or feedback is meh, I'll just move forward with an all in Notifiarr handler or run my local fork. I thought there was some value to ask. Thanks for your thoughts!

_Note: I've not linked the Notifiarr project directly, as I'm directly not affiliated._

Example notifs:

(w/ null fields truncated)
![image](https://github.com/user-attachments/assets/46d59b3c-d03d-49c8-9a9f-f36f61c26583)

![image](https://github.com/user-attachments/assets/b63f2560-d864-44a4-bb3d-1501ccfabdfc)

```
"notifiarr": {
    "class": "opencanary.logger.NotifiarrHandler",
    "webhook_url": "https://notifiarr.com/api/v1/notification/passthrough/**APIKEY**",
    "discord_user_id": ''''discord user id (int)''',
    "discord_color": "00FFFF",
    "discord_chan_id": ''' discord chan id (int)''',
    "time_options": ["local_time_adjusted, "local_time". "utc_time"],
    "drop_empty": boolean
    },
```

**Comment by austinwbest** [comment]
If there is interest in it, native is also an option (assuming the codebase supports it which i guess it does somehow for passthrough to work)

**Comment by clucraft** [comment]
Is this usable in the current state of OpenCanary?
